src/app/_core/services/favorites-cities.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@angular/core';
+import {Router, ActivatedRoute} from "@angular/router";
+import {MAX_FAVORITE_CITY} from "@core/models";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FavoritesCitiesService {
+
+  favoritesCities: string[] =[];
+  starIcon = false;
+
+  constructor(private router:Router,
+              private route:ActivatedRoute,) {}
+
+  onSaveCities(listOfFavoritesCities:string[], cityToAdd:string) {
+    if((!listOfFavoritesCities.includes(cityToAdd)) && listOfFavoritesCities.length<=MAX_FAVORITE_CITY){
```

Comment:
Those 2 conditions needs to be a separated functions `isCityAddedAsFavorite` & `canAddCity`